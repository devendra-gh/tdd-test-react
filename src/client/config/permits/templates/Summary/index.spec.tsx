import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
// import steps from 'client/config/renew-licence/steps';
import SummaryTemplate from './index';

jest.mock('@tamm/app-composer', () => ({
  withTemplateHooks: (component: any) => component,
}));

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('permits/templates/Summary', () => {
  let props: any;

  beforeEach(() => {
    props = {
      permitType: 'Aerial',
      permitInfo: 'test info',
      priceItems: 'test tiem',
      companyDetails: 'Dummy details',
      title: 'title',
      buttons: [{ label: 'test', onClick: jest.fn(), withArrow: false }],
      history: {
        push: jest.fn(),
      },
      actions: {
        title: {
          update: jest.fn(),
        },
        hero: {
          update: jest.fn(),
        },
        breadcrumbs: {
          update: jest.fn(),
        },
      },
      i18n: jest.fn(i => i),
      currentStep: 'DED',
      tableColumns: [
        {
          title: 'testTitle',
        },
      ],
      tableRows: [
        {
          doc: 'testDoc',
        },
      ],
      subTitle: 'test',
      description: 'descriptiontest',
      list: ['test'],
      totalSection: 123,
      onClick: jest.fn(),
    };
  });

  it('should render with all props ', () => {
    const component = shallow(<SummaryTemplate {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render with all props for else condition', () => {
    props.currentStep = '';
    const component = shallow(<SummaryTemplate {...props} />);
    expect(component).toMatchSnapshot();
  });

  // it('should handle button click', () => {
  //   const component = shallow(<SummaryTemplate {...props} />);
  //   const Button = component.find('Button');
  //   const allProps: any = Button.props();
  //   const { onClick } = allProps;
  //   onClick();
  //   expect(props.buttons[0].onClick).toBeCalled();
  // });
});
