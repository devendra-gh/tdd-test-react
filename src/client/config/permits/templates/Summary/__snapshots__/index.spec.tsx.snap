// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`permits/templates/Summary should render with all props  1`] = `
<Container
  sidebar={
    <Connect(Sidebar)
      currentStep="DED"
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "test",
            ],
            Array [
              "descriptiontest",
            ],
            Array [
              "global.aed",
            ],
            Array [
              "test",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": "test",
            },
            Object {
              "type": "return",
              "value": "descriptiontest",
            },
            Object {
              "type": "return",
              "value": "global.aed",
            },
            Object {
              "type": "return",
              "value": "test",
            },
          ],
        }
      }
    />
  }
>
  <div
    className=""
  >
    <h1
      className="syb-title"
      style={
        Object {
          "marginBottom": "2rem",
        }
      }
    >
      test
    </h1>
    <p
      style={
        Object {
          "color": "#000",
        }
      }
    >
      descriptiontest
    </p>
  </div>
  <div
    className="summary"
  >
    <div
      className="summary__group"
    >
      <LocalizedComponent(Table)
        0="t"
        1="e"
        2="s"
        3="t"
        clickable={false}
        commonSelection="unchecked"
        currPage={null}
        disabledSelectionVisible={false}
        editable={false}
        filter={null}
        headerHidden={false}
        i18n={
          [MockFunction] {
            "calls": Array [
              Array [
                "test",
              ],
              Array [
                "descriptiontest",
              ],
              Array [
                "global.aed",
              ],
              Array [
                "test",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": "test",
              },
              Object {
                "type": "return",
                "value": "descriptiontest",
              },
              Object {
                "type": "return",
                "value": "global.aed",
              },
              Object {
                "type": "return",
                "value": "test",
              },
            ],
          }
        }
        isSingleSelect={false}
        items={Array []}
        multiActions={null}
        onActionClick={[Function]}
        onClick={[Function]}
        onFilter={[Function]}
        onPageResize={[Function]}
        onPageTurn={[Function]}
        onRefreshClick={[Function]}
        onSearch={[Function]}
        onSelectionChange={[Function]}
        onSort={[Function]}
        onToggle={[Function]}
        pageResizeOptions={Array []}
        pageSize={null}
        rowVerticalAlign="top"
        search=""
        searchable={false}
        selectable={false}
        singleActions={null}
        size="default"
        sort={null}
        status=""
        title=""
        total={null}
      />
    </div>
    <div
      className="summary__group"
    >
      <LocalizedComponent(Total)
        buttonDisabled={false}
        i18n={[Function]}
        isButtonVisible={false}
        isValueFirst={false}
        onClick={null}
        unit="global.aed"
        value={123}
      />
    </div>
    <div
      className="summary__group"
    >
      <withAnalytics(LocalizedComponent)
        analyticsConfig={
          Object {
            "eventsFrom": Array [],
          }
        }
        aria-label="test"
        label="test"
        onClick={[Function]}
        withArrow={false}
      />
    </div>
  </div>
</Container>
`;

exports[`permits/templates/Summary should render with all props for else condition 1`] = `
<Container
  sidebar={false}
>
  <div
    className=""
  >
    <h1
      className="syb-title"
      style={
        Object {
          "marginBottom": "2rem",
        }
      }
    >
      test
    </h1>
    <p
      style={
        Object {
          "color": "#000",
        }
      }
    >
      descriptiontest
    </p>
  </div>
  <div
    className="summary"
  >
    <div
      className="summary__group"
    >
      <LocalizedComponent(Table)
        0="t"
        1="e"
        2="s"
        3="t"
        clickable={false}
        commonSelection="unchecked"
        currPage={null}
        disabledSelectionVisible={false}
        editable={false}
        filter={null}
        headerHidden={false}
        i18n={
          [MockFunction] {
            "calls": Array [
              Array [
                "test",
              ],
              Array [
                "descriptiontest",
              ],
              Array [
                "global.aed",
              ],
              Array [
                "test",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": "test",
              },
              Object {
                "type": "return",
                "value": "descriptiontest",
              },
              Object {
                "type": "return",
                "value": "global.aed",
              },
              Object {
                "type": "return",
                "value": "test",
              },
            ],
          }
        }
        isSingleSelect={false}
        items={Array []}
        multiActions={null}
        onActionClick={[Function]}
        onClick={[Function]}
        onFilter={[Function]}
        onPageResize={[Function]}
        onPageTurn={[Function]}
        onRefreshClick={[Function]}
        onSearch={[Function]}
        onSelectionChange={[Function]}
        onSort={[Function]}
        onToggle={[Function]}
        pageResizeOptions={Array []}
        pageSize={null}
        rowVerticalAlign="top"
        search=""
        searchable={false}
        selectable={false}
        singleActions={null}
        size="default"
        sort={null}
        status=""
        title=""
        total={null}
      />
    </div>
    <div
      className="summary__group"
    >
      <LocalizedComponent(Total)
        buttonDisabled={false}
        i18n={[Function]}
        isButtonVisible={false}
        isValueFirst={false}
        onClick={null}
        unit="global.aed"
        value={123}
      />
    </div>
    <div
      className="summary__group"
    >
      <withAnalytics(LocalizedComponent)
        analyticsConfig={
          Object {
            "eventsFrom": Array [],
          }
        }
        aria-label="test"
        label="test"
        onClick={[Function]}
        withArrow={false}
      />
    </div>
  </div>
</Container>
`;
